src/SonarScanner.MSBuild.PreProcessor/Interfaces/IPreprocessorObjectFactory.cs

```diff
@@ -43,5 +43,10 @@ public interface IPreprocessorObjectFactory
         /// Creates the component that provisions the Roslyn analyzers.
         /// </summary>
         IAnalyzerProvider CreateRoslynAnalyzerProvider(ISonarWebServer server);
+
+        /// <summary>
+        /// Creates the component that provisions the Roslyn analyzers.
+        /// </summary>
+        IAnalyzerProvider CreateRoslynAnalyzerProvider(ISonarWebServer server, string localCacheTempPath);
```

Comment:
This duplication would make the other overload unused and it would be unclear what to use when and why

Reply:
Yes I was thinking this myself when I made the change, but I wasn't sure if removing the original method would result in a breaking change. If not, I would rather replace that method with the new one I introduced.

Reply:
OK I removed the original method and reflected the change in the tests. It appears that the tests are passing.